braindecode/preprocessing/eegprep_preprocess.py

```diff
@@ -0,0 +1,1139 @@
+    behavior, for example if you're building an EEGPrep-like pipeline from individual
+    steps, e.g., to customize parts that are not exposed by the top-level EEGPrep
+    preprocessor.
+
+    NOTE: If your method involves causal analysis, either with applications to real-time
+    single-trial brain-computer interfacing or for example involving autoregressive
+    modeling or other causal measures, consider using a strictly causal highpass filter
+    instead.
+
+    Parameters
+    ----------
+    transition : Sequence[float]
+        The transition band in Hz, i.e. lower and upper edge of the transition as in
+        (lo, hi). Defaults to (0.25, 0.75). Choosing this can be tricky when your data
+        contains long-duration event-related potentials that your method exploits, in
+        which case you may need to carefully lower this somewhat to avoid attenuating
+        them.
+    attenuation : float
+        The stop-band attenuation, in dB. Defaults to 80.0.
+    method : str
+        The method to use for filtering ('fft' or 'fir'). Defaults to 'fft' (uses more
+        memory but is much faster than 'fir').
+
+    """
+
+    def __init__(
+        self,
+        transition: Sequence[float] = (0.25, 0.75),
+        *,
+        attenuation: float = 80.0,
+        method: str = "fft",
+    ):
+        super().__init__()
+        self.transition = transition
+        self.attenuation = attenuation
+        self.method = method
+
+    def apply_eeg(self, eeg: dict[str, Any], raw: BaseRaw) -> dict[str, Any]:
+        """Apply the preprocessor to an EEGLAB EEG structure. Overridden by subclass."""
+        eeg = eegprep.clean_drifts(
+            eeg,
+            transition=self.transition,
+            attenuation=self.attenuation,
+            method=self.method,
+            # (no other args)
+        )
+
+        return eeg
+
+
+class Resampling(EEGPrepBasePreprocessor):
+    """Resample the data to a specified rate. Included to equivalence to EEGPrep.
+
+    MNE has its resampling routine (use as `Preprocessor("resample", sfreq=rate)`)
+    but this will not necessarily match EEGPrep's behavior exactly; to get bit-identical
+    results to the default EEGPrep pipeline, use this preprocessor instead.
+
+    Parameters
+    ----------
+    sfreq : float | None
+        The desired sampling rate in Hz. Skipped if set to None.
+
+    """
+
+    def __init__(
+        self,
+        sfreq: float | None,
+    ):
+        super().__init__(can_change_duration=True)
+        self.sfreq = sfreq
+
+    def apply_eeg(self, eeg: dict[str, Any], raw: BaseRaw) -> dict[str, Any]:
+        """Apply the preprocessor to an EEGLAB EEG structure. Overridden by subclass."""
+        if self.sfreq is not None:
+            eeg = eegprep.resample(eeg, self.sfreq)
+
+        return eeg
+
+
+class RemoveBadChannels(EEGPrepBasePreprocessor):
+    """Removes EEG channels with problematic data; variant that uses channel locations.
+
```

Comment:
Can you put a figure example link please? it would be also necessary when I create the docstring for the eegprep. 

```suggestion
    .. figure:: https://....
       :align: center
       :alt: Remove Bad Channels
      
      Caption
```